lib/stack_master/parameter_resolvers/stack_output.rb

```diff
@@ -10,14 +10,14 @@ def initialize(config, stack_definition)
         @config = config
         @stack_definition = stack_definition
         @stacks = {}
+        @output_regex = %r{(?:([^:]+):)?([^:/]+)/(.+)}
       end
 
       def resolve(value)
-        validate_value!(value)
-        stack_name, output_name = value.split('/')
-        stack = find_stack(stack_name)
+        region, stack_name, output_name = parse!(value)
```

Comment:
It might be worth defaulting `region` to the current region to have a single path/logic resolving stack outputs and ensuring `region` will never be `nil`, simplifying all the logic.